src/typechecker/FStar.TypeChecker.Normalize.fs

```diff
@@ -1447,13 +1449,7 @@ and do_unfold_fv cfg env stack (t0:term) (qninfo : qninfo) (f:fv) : term =
          log_unfolding cfg (fun () -> BU.print2 " >> Unfolded %s to %s\n"
                        (Print.term_to_string t0) (Print.term_to_string t));
          // preserve the range info on the returned term
-         let t =
-           if cfg.steps.unfold_until = Some delta_constant
-           //we're really trying to compute here; no point propagating range information
-           //which can be expensive
-           then t
-           else Subst.set_use_range t0.pos t
-         in
+         let t = Subst.set_use_range t0.pos t in
```

Comment:
This diff is my main concern ...

The previous code was specifically sacrificing propagating the use range since it is expensive in compute-heavy reductions.

So, I'm concerned about the performance cost here.

Reply:
I have been looking around the code and I can't figure out another way to update the use range. Is there a way to update the use range of the result of the computation without propagating it through each computation step?